### PR TITLE
Add displayName, description and category to template schema

### PR DESCRIPTION
**Pull request by samutoljamo-pareto**
I used "description" instead of "templateDescription". Since we are already dealing with templates, I do not think the "template" in the variable name adds any clarity.  Description, displayName and category are optional

### CODE REVIEW

File: tools/data-handler/src/show.ts [hunk 1]
```diff
@@ -281,9 +281,14 @@ export class Show {
      * @todo: add unit tests
      * @returns all templates in a project.
      */
-    public async showTemplatesWithDetails(projectPath: string): Promise<(resource)[]> {
+    public async showTemplatesWithDetails(projectPath: string): Promise<template[]> {
         Show.project = new Project(projectPath);
-        return await Show.project.templates();
+        const promiseContainer = [];
```
Comment (arime-pareto) [inline tools/data-handler/src/show.ts:286]: Suggestion: could it be cleaner to use map over templates instead a for loop and push?

File: tools/data-handler/src/show.ts [hunk 2]
```diff
@@ -281,9 +281,14 @@ export class Show {
...
+        for (const template of await Show.project.templates()) {
+            promiseContainer.push(Show.project.createTemplateObjectByName(template.name).then(t => t?.show()));
+        }
+        const result = await Promise.all(promiseContainer);
+        return result.filter(item => item) as template[];
```
Comment (arime-pareto) [inline tools/data-handler/src/show.ts:291]: This is fine, but a nice trick to filter undefined values from an array is to use `result.filter(Boolean)`.

**Review by arime-pareto — APPROVED** [review]
Looks good to me, left just a couple of improvement suggestions.

**Review by arime-pareto — APPROVED** [review]

File: tools/data-handler/src/show.ts (hunk 1 above)
Comment (samutoljamo-pareto) [reply tools/data-handler/src/show.ts]: Yes it would, but the same syntax is also used elsewhere. I replace the for loop with map in this one

**Pull request merged**